spec/datadog/data_streams/processor_spec.rb

```diff
@@ -22,16 +22,25 @@
   let(:agent_settings) { Datadog::Core::Configuration::AgentSettings.new(adapter: :test, hostname: 'localhost', port: 9999) }
   let(:processor) { described_class.new(interval: 10.0, logger: logger, settings: settings, agent_settings: agent_settings) }
 
+  after do
+    processor.stop(true)
+  end
```

Comment:
Minor: If the background thread shutdown takes too long, `stop` returns _without_ waiting for `join` to finish, and thus the thread could potentially still remain around for a little bit more, triggering the leak detection. (I use the same pattern in [the profiler specs](https://github.com/DataDog/dd-trace-rb/blob/6eff05b1f2ddb654189660724867fabe9adacdfe/spec/datadog/profiling/scheduler_spec.rb#L41))

```suggestion
  after do
    processor.stop(true)
    processor.join
  end
```